sound/soc/sof/amd/acp.c

```diff
@@ -284,9 +284,22 @@ static irqreturn_t acp_irq_thread(int irq, void *context)
 
 	val = snd_sof_dsp_read(sdev, ACP_DSP_BAR, ACP_DSP_SW_INTR_STAT);
 	if (val & ACP_DSP_TO_HOST_IRQ) {
+		while (snd_sof_dsp_read(sdev, ACP_DSP_BAR, ACP_AXI2DAGB_SEM_0)) {
+			/* Wait until acquired HW Semaphore lock or timeout */
+			timeout--;
+			if (!timeout) {
+				dev_err(sdev->dev, "Failed to acquire HW lock\n");
```

Comment:
nit-pick: when you have two identical logs, it's difficult to figure out which one happened. Consider using something unique e.g. __func__ to make the difference.